### PR TITLE
tests: use a `server` fixture

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

**Comment by Saviq** [comment]
bors try

File: frame/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 inotify
-mypy
 pytest
 asciinema
 pywayland
+pytest
```
Comment (wmww) [inline frame/requirements.txt:5]: duplicate `pytest` entry

File: frame/conftest.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return cmd
+
+@pytest.fixture(scope='session')
+def mypy(request) -> types.ModuleType:
+    deps: bool = request.config.getoption("--deps", False)
+    try:
+        mypy = __import__('mypy.api')
+    except ModuleNotFoundError:
+        if deps:
+            subprocess.check_call(('pip', 'install', 'mypy'))
```
Comment (wmww) [inline frame/conftest.py:68]: why do we install mypy this way instead of in requirements.txt?

File: frame/test_apps_can_run.py [hunk 3]
```diff
@@ -1,12 +1,11 @@
 from display_server import DisplayServer
-from helpers import all_servers
 import pytest
 import time
 
 short_wait_time = 3
 
 class TestAppsCanRun:
-    @pytest.mark.parametrize('server', all_servers())
```
Comment (wmww) [inline frame/test_apps_can_run.py:9]: were is the value of `server` coming from without this?

**Review by wmww — CHANGES_REQUESTED** [review]
I'm a bit confused we'll chat at sync tomorrow

File: frame/conftest.py (hunk 2 above)
Comment (Saviq) [reply frame/conftest.py:66]: As discussed - because we only run a subset of tests, so we do this to avoid installing unnecessary dependencies.

**Review by wmww — APPROVED** [review]
LGTM
bors r+

**Pull request merged**